### PR TITLE
import fg_json to easyrec_config

### PR DESCRIPTION
**Pull request by dawn310826**

### CODE REVIEW

File: easy_rec/python/train_eval.py [hunk 1]
```diff
@@ -73,6 +74,9 @@ def main(argv):
       else:
         assert FLAGS.ignore_finetune_ckpt_error, 'fine_tune_checkpoint(%s) is not exists.' % FLAGS.fine_tune_checkpoint
 
+    if pipeline_config.fg_json_path:
+      fg_util.load_fg_json_to_config(pipeline_config)
```
Comment (tiankongdeguiji) [inline easy_rec/python/train_eval.py:78]: It’s better to load_fg_json_to_config in train_and_evaluate_impl.

File: easy_rec/python/utils/convert_rtp_fg.py [hunk 2]
```diff
@@ -201,6 +207,34 @@ def convert_rtp_fg(rtp_fg,
       else:
         pipeline_config.feature_config.features.append(feature_config)
       pipeline_config.data_config.input_fields.append(input_field)
+
+      if 'extra_combo_info' in feature:
```
Comment (tiankongdeguiji) [inline easy_rec/python/utils/convert_rtp_fg.py:211]: The operation `Combo` has been done in FG, why should we convert ComboFeature here? simply treat it as an id_feature.

File: samples/rtp_fg/fg_test_extensions.config [hunk 3]
```diff
@@ -0,0 +1,111 @@
+train_input_path: "../../data/test/rtp/taobao_train_feature.txt"
+eval_input_path: "../../data/test/rtp/taobao_test_feature.txt"
+model_dir: "experiments/fg_test_extensions"
+fg_json_path: './samples/rtp_fg/fg_test_extensions.json'
```
Comment (tiankongdeguiji) [inline samples/rtp_fg/fg_test_extensions.config:4]: where is the test?

**Review by tiankongdeguiji — CHANGES_REQUESTED** [review]

File: easy_rec/python/train_eval.py (hunk 1 above)
Comment (dawn310826) [reply easy_rec/python/train_eval.py:78]: This is to make the priority of 'load_fg_json_to_config' less than 'load hpo_params' and 'edit_config_json'.

File: easy_rec/python/utils/fg_util.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+from easy_rec.python.protos.dataset_pb2 import DatasetConfig
+from easy_rec.python.utils.config_util import get_compatible_feature_configs
+from easy_rec.python.utils.convert_rtp_fg import convert_rtp_fg
+
+def load_fg_json_to_config(pipeline_config):
+  fg_json_path = pipeline_config.fg_json_path
+  if not fg_json_path:
+    return
+  label_fields = pipeline_config.data_config.label_fields
+  fg_config = convert_rtp_fg(fg_json_path, label_fields=label_fields)
```
[the remark this answers is not in the document]
Comment (dawn310826) [reply easy_rec/python/utils/fg_util.py]: done

File: easy_rec/python/test/fg_test.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    super(FGTest, self).__init__(methodName=methodName)
+    self._pipeline_config_path = 'samples/rtp_fg/fg_test_extensions.config'
+    self._fg_path = 'samples/rtp_fg/fg_test_extensions.json'
+
+  def test_fg_json_to_config(self):
+    pipeline_config = config_util.get_configs_from_pipeline_file(
+        self._pipeline_config_path)
+    pipeline_config.fg_json_path = self._fg_path
+    fg_util.load_fg_json_to_config(pipeline_config)
+    assert len(pipeline_config.data_config.input_fields) == 19 and \
```
Comment (tiankongdeguiji) [inline easy_rec/python/test/fg_test.py:21]: assert equal to a pbtext and use self.assertEqual

File: easy_rec/python/utils/convert_rtp_fg.py (hunk 2 above)
Comment (tiankongdeguiji) [reply easy_rec/python/utils/convert_rtp_fg.py:176]: add extra_combo_info in fg doc

File: easy_rec/python/test/fg_test.py (hunk 5 above)
Comment (dawn310826) [reply easy_rec/python/test/fg_test.py]: done

File: easy_rec/python/utils/convert_rtp_fg.py (hunk 2 above)
Comment (dawn310826) [reply easy_rec/python/utils/convert_rtp_fg.py:176]: done

**Pull request merged**